src/app/icd/ICDClientInfoManagement.h

```diff
@@ -0,0 +1,116 @@
+/*
+ *    Copyright (c) 2023 Project CHIP Authors
+ *    All rights reserved.
+ *
+ *    Licensed under the Apache License, Version 2.0 (the "License");
+ *    you may not use this file except in compliance with the License.
+ *    You may obtain a copy of the License at
+ *
+ *        http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *    Unless required by applicable law or agreed to in writing, software
+ *    distributed under the License is distributed on an "AS IS" BASIS,
+ *    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *    See the License for the specific language governing permissions and
+ *    limitations under the License.
+ */
+
+/**
+ *    @file
+ *      This file defines a basic implementation of ICDClientInfosPersistentStorage that
+ *      persists subscriptions in a flat list in TLV.
+ */
+
+#pragma once
+
+#include <app/icd/ICDClientInfoPersistentStorage.h>
+#include <lib/core/CHIPCore.h>
+
+#include <lib/core/TLV.h>
+#include <lib/support/DefaultStorageKeyAllocator.h>
+#include <lib/support/Pool.h>
+
+namespace chip {
+namespace Crypto {
+using SymmetricKeystore = SessionKeystore;
+}
+} // namespace chip
+
+namespace chip {
+namespace app {
+
+/**
+ * An ICDClientInfoManagement using PersistentStorageDelegate as it backend.
+ */
+class ICDClientInfoManagement : public ICDClientInfoPersistentStorage
+{
+public:
+    static constexpr size_t kIteratorsMax = CHIP_CONFIG_MAX_ICD_CLIENTS_INFO_STORAGE_CONCURRENT_ITERATORS;
+
+    /**
+     * @brief Retrieve the singleton ICDClientInfoManagement.
+     *
+     *  @return  A pointer to the shared ICDClientInfoManagement
+     *
+     */
+    static ICDClientInfoManagement * GetInstance(void);
```

Comment:
Does this need to be a singleton? Could it be owned by the "server" equivalent for controllers?

Reply:
this would be shared/accessed by multiple controller...it means it needs init before controller init..it seems we don't have such equivalent thing in client side..